### PR TITLE
test: setup azure pipeline and add e2e

### LINKED ISSUE #56: Switch to using go modules

### PR DESCRIPTION
**Pull request by aramase**
- [x] Adds azure pipeline to run tests
- [x] Removes pipeline for circle CI
- [x] Switches to using go modules

fixes #56

### CODE REVIEW

**Comment by aramase** [comment]
Need to remove `circle CI` build after this PR is merged.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "go.inferGopath": false
```
Comment (ritazh) [inline .vscode/settings.json:2]: Was this file intentionally checked in?

**Comment by ritazh** [comment]
just one nit. Everything else LGTM

File: .vscode/settings.json (hunk 1 above)
Comment (aramase) [reply .vscode/settings.json:2]: Removed the file and added it to `.gitignore`

**Review by ritazh — APPROVED** [review]
lgtm

**Pull request merged**